logstash-core/lib/logstash/outputs/base.rb

```diff
@@ -63,10 +63,7 @@ def self.plugin_type
   def initialize(params={})
     super
     config_init(@params)
-
-    # If we're running with a single thread we must enforce single-threaded concurrency by default
-    # Maybe in a future version we'll assume output plugins are threadsafe
-    @single_worker_mutex = Mutex.new
+    @receives_encoded = self.methods.include?(:multi_receive_encoded)
```

Comment:
Since this is called once per batch I wanted to only calculate this once.

I could make a method with `||=` but that seems like adding complexity.